cli/tools/pm/cache_deps.rs

```diff
@@ -94,7 +100,17 @@ pub async fn cache_top_level_deps(
             });
           }
         }
-        "npm" => roots.push(specifier.clone()),
+        "npm" => {
+          let Ok(req_ref) =
+            NpmPackageReqReference::from_str(specifier.as_str())
+          else {
+            continue;
+          };
+          if workspace_npm_package_names.contains(&*req_ref.req().name) {
```

Comment:
I kind of wonder if we should still install when the req doesn't match the version in the package.json, but this is probably fine.